### ISSUE TITLE
The Javascript suffix in the preference window doesn't handle periods.

### ISSUE THREAD

**Issue by mishunov** [bdkjones/CodeKit-1-Old#2]

**Comment by bdkjones** [comment]
After careful consideration, I've decided to mark this "behaves as intended."

If you have a file called "myAwesomeScript-min.js" and you've set the output suffix in CodeKit to be "-min", then you'll see that your javascript file shows the following output path:

"myAwesomeScript-min-min.js"

I COULD change the app to recognize that this file is already minified (by looking for your supplied suffix). But then, if you choose to process the file, CodeKit will overwrite the source file (myAwesomeScript.min.js) with the output file (myAwesomeScript.min.js)

CodeKit (and less.app) have ONE cardinal rule: Never destroy data.

In this case, suppose some idiot names his javascript file "script.min.js" but the script itself is NOT a minified one. That is, it has comments and structure and the whole nine yards. Well, if we changed CodeKit to the above behavior, it would think this is a minified file and it would overwrite it with the new minified version whenever the user clicked "process." BOOM: destroyed data. Comments are gone, structure is gone, etc.

So the safe bet is to show in the UI what happens if you try to process an "already minified" file: We'll process it again and, because we NEVER overwrite the source file, we'll tack on an extra suffix and let you sort it out.

The reason this is not an issue for CSS files is that CodeKit does not add CSS files to the app. It does not recognize them. 

Bottom line: I want to err on the side of caution and make sure the app is completely incapable of overwriting source files. However, I have changed the logic to automatically ignore any javascript file that contains the "minified" suffix set in the preferences and I'm considering the suggestion to "dim" ignored files to make it more obvious that you probably don't want to process this javascript file.
